### ISSUE TITLE
[BUG] Dungeon Finder Triggering Hourly Cap

### ISSUE THREAD

**Issue by Raxxed** [JamminL/cata-classic-bugs#1217]
#### 1.1. Description
If you enter 5 dungeons via the Dungeon Finder within 60 minutes, then you are prevented from entering any further dungeons via the Dungeon Finder until the hourly cap expires. This is not correct behavior, the Dungeon Finder should ignore the hourly cap.

  #### 1.2. How to Reproduce
          {Describe the steps to reproduce the particular bug, in detail:
1. Specific queue and then either complete or leave 5 dungeons within an hour (You can queue as a 5man premade and instantly leave specific queued dungeons without any penalty)
2. If you attempt to Specific Queue for a dungeon, after having queued into 5 dungeons within 60 minutes, the "Find Group" or "Join as Group" button simply does nothing. There are no UI errors, no feedback, the button just does nothing.
3. If you attempt to Random queue for a dungeon, you will be able to successfully queue, however once the group has formed and you are supposed to be teleported into the instance, you will instead fail to teleport in and receive the "You have entered too many instances recently" error. 
           
  #### 1.3  Source Material
  My 5man group has encountered this issue when attempting to speed level to 85 on the beta by spamming Blackrock Caverns. We then attempted to replicate it on live (Cata prepatch) and we were able to do so using currently available dungeons.
#### 2.1. Description
The dungeon finder does not care about the 5 dungeon per hour limit. It never has at any point in the past of either Classic or Retail since it's original implementation. 
  #### 2.2 Source Material
{Video, Posts, etc..}

This has the potential to be catastrophic for Cataclysm Launch given the amount of players that prefer to dungeon grind than quest to max level on launch.

**Comment by oldspices** [comment]
it seems to happen sporadically, as ive done more than 5 and had no problem then randomly ran into problems on other sessions.  I made a report here https://github.com/JamminL/cata-classic-bugs/issues/1149

**Comment by Raxxed** [comment]
> it seems to happen sporadically, as ive done more than 5 and had no problem then randomly ran into problems on other sessions. I made a report here #1149

The behavior seems extremely strange, as you are able to manually zone into other dungeons despite being capped, at least from our experience. Just not the dungeons that contributed to the 5 dungeon cap 

We did 5 BRC runs, triggered the above mentioned lockout bug but we could manually run into RFC no problem, we even flew down to Vortex Pinnacle to check if it was specifically Cata dungeons were broken but we could zone into that too (but we couldn't manually zone into BRC). According to NIT we'd entered 9 dungeons within 60 minutes (5 RDF, 4 manual).

**Comment by Raxxed** [comment]
I have done some further testing and can consistently reproduce this with the exact same result. 

Steps:
1. Run 5 dungeons, these can be the same dungeon, you can just manually run in/out and reset to lock yourself. Confirm that you can no longer zone into the dungeon. 
2. Pick a dungeon that you are level-eligible to specific queue for.
3. Specific queue for this dungeon - you will be able to, despite technically being locked out. 
4. Decline the queue and manually enter the dungeon you chose **OR** accept the queue for the dungeon and complete it, so long as you have that dungeon on your "recently entered" list. Either way works. 
5. Attempt to specific queue for this dungeon again. Congrats, you have broken your Dungeon Finder tool. The Find Group button will now do nothing when you click it, regardless of if you queue for something else entirely.
6. Relog
7. Attempt to specific queue for a **DIFFERENT** dungeon that you do not have a "recently entered" entry for. It will allow you to successfully queue. 
8. Return to step 4.

tl;dr - this bug only applies to dungeons that you have entered in the last 60 minutes.

**Comment by oldspices** [comment]
![image](https://github.com/JamminL/cata-classic-bugs/assets/26885524/792fa3aa-b8a4-4075-bf9d-434681ce3c1e)
 Is it maybe 10 in 1 hour? i just got this and i did 1 or 2 specifics then like 8 randoms in a row.... then got hit

**Comment by Raxxed** [comment]
> ![image](https://private-user-images.githubusercontent.com/26885524/330614646-792fa3aa-b8a4-4075-bf9d-434681ce3c1e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3Mjg2MzAsIm5iZiI6MTcxNTcyODMzMCwicGF0aCI6Ii8yNjg4NTUyNC8zMzA2MTQ2NDYtNzkyZmEzYWEtYjhhNC00MDc1LWJmOWQtNDM0NjgxY2UzYzFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE0VDIzMTIxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxZTEwOGYyMWI5Njc2YTBiNmViYzUzZGQ4YjUzNzQ5MGVhOGRlNWI3MWY3YmM2OGM0ZDljY2FhOTdiM2UxZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5pdFoiNXyy0qqzxW4BSpMT5_oq6dO6pD4xGHymyT7fg) Is it maybe 10 in 1 hour? i just got this and i did 1 or 2 specifics then like 8 randoms in a row.... then got hit

I think the Dungeon Finder is a red herring here and the actual issue is with dungeon lockouts themselves, it's just causing the Dungeon Finder to break. Here's a quick test which bypasses the dungeon finder that works for me consistently that you can try too.

Head to HOL/HOS, Zone in/out of one dungeon 5 times resetting it until you hit the 5 instance limit, it will stop you going back in after 5 as intended, but if you then fly over to the other dungeon and attempt to zone in, it will let you zone in, BUT if you then zone out, reset and zone back in you will be locked. If you want you can then go and find another unrelated dungeon to zone into and you should experience the same thing. Not locked > reset > locked.
